internal/engine/evaluator.go

```diff
@@ -160,13 +161,9 @@ func (rpe *resourcePolicyEvaluator) Evaluate(ctx context.Context, tctx tracer.Co
 					octx.Skipped(err, "Error evaluating output")
 				}
 
-				ruleName := rule.Name
-				if ruleName == "" {
-					ruleName = fmt.Sprintf("rule-%03d", i)
-				}
 				result.Outputs = append(result.Outputs, &enginev1.OutputEntry{
-					Src: fmt.Sprintf("%s#%s", rpe.policy.Meta.Fqn, ruleName),
-					Val: output,
+					Src: namer.RuleFQN(rpe.policy.Meta, p.Scope, rule.Name),
+					Val: structpb.NewStringValue(output),
```

Comment:
What I meant here about using protobuf `Value` is that evaluating the output expression could produce any kind of value, not just strings. For example, if the output expr is `1+1`, then that should produce a `NumberValue`. That way, the response would be typed rather than everything being strings.

Reply:
🤦‍♂️